### PR TITLE
fs/fssyscall: change Open()'s signature to mimic syscall.Open()

### PR DESCRIPTION
**Pull request by amery**
BREAKING

## Summary by CodeRabbit

- **New Features**
	- Enhanced file opening flexibility by allowing specification of mode and permissions in the file system operations.

- **Bug Fixes**
	- Improved error handling clarity in file locking operations.

### CODE REVIEW

File: fs/flock/flock.go [hunk 1]
```diff
@@ -2,19 +2,22 @@
...
 import (
+	"os"
+
 	syscall "darvaza.org/x/fs/fssyscall"
 )
 
 // LockEx locks a file by name
 func LockEx(filename string) (syscall.Handle, error) {
-	zero := syscall.ZeroHandle
+	const zero = syscall.ZeroHandle
```
[the remark this answers is not in the document]
Comment (karasz) [reply fs/flock/flock.go:12]: We are aligning the `syscal.Open()` function not `LockEx`

**Review by karasz — APPROVED** [review]
LGTM

**Comment by amery** [comment]
> LGTM

Yes, LockEx is good with a single argument I believe. syscall.Open() is the one needing mode and permissions

**Comment by amery** [comment]
`darvaza.org/x/fs@v0.3.2` tagged

**Pull request merged**